repo/manifest/committed_manifest_manager.go

```diff
@@ -137,20 +137,45 @@ func (m *committedManifestManager) loadCommittedContentsLocked(ctx context.Conte
 	m.verifyLocked()
 
 	var (
-		mu        sync.Mutex
-		manifests map[content.ID]manifest
+		mu sync.Mutex
+
+		// Temporary set of mappings that we can swap with what's currently in the
+		// manager once we're done. This ensures we don't clobber existing data if
+		// we fail to load manifests.
+		committedEntries    = map[ID]*manifestEntry{}
+		committedContentIDs = map[content.ID]bool{}
 	)
 
 	for {
-		manifests = map[content.ID]manifest{}
-
 		err := m.b.IterateContents(ctx, content.IterateOptions{
 			Range:    index.PrefixRange(ContentPrefix),
 			Parallel: manifestLoadParallelism,
 		}, func(ci content.Info) error {
-			man, err := loadManifestContent(ctx, m.b, ci.GetContentID())
+			mu.Lock()
+			committedContentIDs[ci.GetContentID()] = true
+			mu.Unlock()
+
+			err := forEachManifestEntry(
+				ctx,
+				m.b,
+				ci.GetContentID(),
+				func(e *manifestEntry) bool {
+					mu.Lock()
+					prev := committedEntries[e.ID]
+					mu.Unlock()
+
+					if prev == nil || e.ModTime.After(prev.ModTime) {
+						mu.Lock()
+						committedEntries[e.ID] = e
+						mu.Unlock()
+					}
+
+					// Continue iteration.
+					return true
```

Comment:
Just for my understanding, we are not enabling partial deserialization in this PR, right?

Reply:
correct, even though the PR adds the code to deserialize just some of the entries, we continue to deserialize everything since this is the only place that codepath is called and this still needs to examine all manifests

Reply:
got it, thanks!